lib/rspec/matchers/built_in/respond_to.rb

```diff
@@ -23,6 +26,43 @@ def with(n)
         end
 
         # @api public
+        # Specifies keyword arguments, if any.
+        #
+        # @example
+        #   expect(obj).to respond_to(:message).with_keywords(:color, :shape)
+        # @example with an expected number of arguments
+        #   expect(obj).to respond_to(:message).with(3).arguments.and_keywords(:color, :shape)
+        def with_keywords(*keywords)
+          @expected_keywords = keywords
+          self
+        end
+        alias :and_keywords :with_keywords
```

Comment:
I guess it's ok that technically you could to `respond_to(:message).and_keywords(:color, :shape)`